test/ejabberd_tests/tests/mod_http_upload_SUITE.erl

```diff
@@ -0,0 +1,211 @@
+
+-define(NS_HTTP_UPLOAD, <<"urn:xmpp:http:upload">>).
+-define(S3_HOSTNAME, "http://bucket.s3-eu-east-25.example.com").
+-define(S3_OPTS,
+        [
+         {s3, [
+               {bucket_url, ?S3_HOSTNAME},
+               {region, "eu-east-25"},
+               {access_key_id, "AKIAIAOAONIULXQGMOUA"},
+               {secret_access_key, "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"}
+              ]}
+        ]).
+
+%%--------------------------------------------------------------------
+%% Suite configuration
+%%--------------------------------------------------------------------
+
+all() ->
+    [{group, http_upload_s3}].
+
+groups() ->
+    [{http_upload_s3, [], [
+                           http_upload_service_discovery,
+                           request_slot,
+                           get_url_ends_with_filename,
+                           urls_contain_s3_hostname,
+                           rejects_empty_filename,
+                           rejects_negative_filesize,
+                           rejects_invalid_size_type
+                          ]}].
+
+suite() ->
+    escalus:suite().
+
+%%--------------------------------------------------------------------
+%% Init & teardown
+%%--------------------------------------------------------------------
+
+init_per_suite(Config) ->
+    escalus:init_per_suite(Config).
+
+end_per_suite(Config) ->
+    escalus:end_per_suite(Config).
+
+init_per_group(http_upload_s3, Config) ->
+    dynamic_modules:start(<<"localhost">>, mod_http_upload, ?S3_OPTS),
+    escalus:create_users(Config, escalus:get_users([bob])).
+
+end_per_group(_, Config) ->
+    dynamic_modules:stop(<<"localhost">>, mod_http_upload),
+    escalus:delete_users(Config, escalus:get_users([bob])).
+
+init_per_testcase(CaseName, Config) ->
+    escalus:init_per_testcase(CaseName, Config).
+
+end_per_testcase(CaseName, Config) ->
+    escalus:end_per_testcase(CaseName, Config).
+
+%%--------------------------------------------------------------------
+%% Service discovery test
+%%--------------------------------------------------------------------
+
+http_upload_service_discovery(Config) ->
+    escalus:story(
+      Config, [{bob, 1}],
+      fun(Bob) ->
+              ServJID = escalus_client:server(Bob),
+              Result = escalus:send_and_wait(Bob, escalus_stanza:disco_info(ServJID)),
+              escalus:assert(is_iq_result, Result),
+              escalus:assert(has_feature, [?NS_HTTP_UPLOAD], Result)
+      end).
+
+request_slot(Config) ->
+    escalus:story(
+      Config, [{bob, 1}],
+      fun(Bob) ->
+              ServJID = escalus_client:server(Bob),
+              Request = create_slot_request_stanza(ServJID, <<"filename.jpg">>, 123, undefined),
+              Result = escalus:send_and_wait(Bob, Request),
+              escalus:assert(is_iq_result, Result),
+              escalus:assert(fun check_namespace/1, Result),
+              escalus:assert(fun check_put_and_get_presence/1, Result)
```

Comment:
I'd call it `check_put_and_get_presence` because it implies a bit that `<presence>` appears somewhere here. Maybe `check_put_and_get_fields`?